### ISSUE TITLE
CA1305 false positives: `Convert.ToString(bool)`, `Convert.ToChar(string)`, `Convert.ToBoolean(strng)`

### ISSUE THREAD

**Issue by MaceWindu** [dotnet/roslyn-analyzers#7154]
### Analyzer

**Diagnostic ID**: [CA1305](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305): `Specify IFormatProvider`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.Analyzers)

**Version**: 3.11.0-beta1.23525.2 (Latest)

### Describe the bug

CA1305 still complains about some methods that doesn't benefit from `IFormatProvider`

### Steps To Reproduce

```cs
Convert.ToString(true);
Convert.ToChar("x");
Convert.ToBoolean("true")
```

### Expected behavior

methods should be ignored by analyzer

### Actual behavior

CA1305 generated
